aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/MutationQueue.java

```diff
@@ -27,8 +27,11 @@
  * The {@link MutationQueue} is a LinkedHashMap like container , the goal of using this container is to
  * achieve O(1) time complexity for both getting a {@link PendingMutation} and update an existing mutation with
  * valid id.
- * MutationQueue is implementing the Queue interface and provide most of the queue operations,
+ * MutationQueue is implementing the Queue interface and provide most of the queue operations.
+ * 
+ * @deprecated This class was released with public visibility but is not intended to be consumed.
  */
+@Deprecated
```

Comment:
Let's add a note here that this will be removed from a future version of Amplify.